File: apps/net/Ch3_Lang/L03_FunctionBasics.cs
using System;


namespace Ch3_Lang
{
    class FunctionBasics
    {
        public static void Run()
        {
            ShowHeader();

            Random rand = new Random();
            int theNumber = rand.Next(1, 100);
            int count = 0;

            while (true)
            {
                int? guess = GetGuess();
                if (guess == null)
                {
                    continue;
                }

                count++;
                if (EvaluateGuess(guess.Value, theNumber))
                {
                    break;
                }
            }

            Console.WriteLine($"You got the number in {count} attempts. Thanks for playing, bye.");
        }

        private static void ShowHeader()
        {
            Console.WriteLine("-------------------------------------------");
            Console.WriteLine("|                                         |");
            Console.WriteLine("|           C# HIGH / LOW GAME            |");
            Console.WriteLine("|                                         |");
            Console.WriteLine("-------------------------------------------");
            Console.WriteLine();
            Console.WriteLine("I'm thinking of a number between 1 & 100.");
            Console.WriteLine("How many steps can you guess it in?");
            Console.WriteLine();
        }

        private static int? GetGuess()
        {
            Console.Write("What number am I thinking of? ");
            string text = Console.ReadLine();
            if (!int.TryParse(text, out int val))
            {
                Console.WriteLine("That's not a number!");
                return null;
            }

            if (val < 1 || 100 < val)
            {
                Console.WriteLine($"{val} is not between 1 & 100.");
                return null;
            }

            return val;
        }


        private static bool EvaluateGuess(int guess, int number)
        {
            if (guess == number)
            {
                Console.WriteLine($"That's it! I was thinking of {number}.");
            }
            else if (guess < number)
            {
                Console.WriteLine("That's too LOW");
            }
            else
            {
                Console.WriteLine("That's too HIGH");
            }

            return guess == number;
        }
    }
}

File: apps/py/ch03_lang/L03_function_basics.py
import random


def main():
    show_header()

    # Random rand =new Random();
    # the_number = rand.Next(1, 100);
    the_number = random.randint(1, 100)

    count = 0
    while True:
        guess = get_guess()
        if not guess:
            continue

        count += 1
        if evaluate_guess(guess, the_number):
            break

    print(f"You got the number in {count} attempts. Thanks for playing, bye!")


def evaluate_guess(guess, number):
    if guess == number:
        print(f"That's it! I was thinking of {number}.")
    if guess < number:
        print("That's too LOW")
    elif guess > number:
        print("That's too HIGH")

    return guess == number


def get_guess():
    val = None
    try:
        text = input("What number am I thinking of? ")
        val = int(text)
        if val < 1 or 100 < val:
            print(f"{val} is not between 1 and 100.")
            return None
        return val
    except:
        print(f"{val} is not an integer!")
        return None


def show_header():
    print('-------------------------------------------')
    print('|                                         |')
    print('|         PYTHON HIGH / LOW GAME          |')
    print('|                                         |')
    print('-------------------------------------------')
    print()
    print("I'm thinking of a number between 1 & 100. ")
    print("How many steps can you guess it in?")
    print()

    # implicit
    return None


if __name__ == '__main__':
    main()
